libkbfs/folder_branch_ops.go

```diff
@@ -4135,6 +4136,14 @@ func (fbo *folderBranchOps) syncAllLocked(
 				if !fbo.nodeCache.IsUnlinked(newNode) {
 					resolvedPaths[newPointer] = newPath
 				}
+				cleanups = append(cleanups,
+					func(ctx context.Context, lState *lockState, err error) {
+						if err != nil {
+							return
+						}
+						fbo.blocks.ClearCachedDirEntry(lState, newPath)
+						fbo.status.rmDirtyNode(newNode)
+					})
```

Comment:
The place that calls these cleanup functions don't have access to `newPath` and `newNode` though, hence the need for the closures.  Maybe you're suggesting storing those in a slice, and then processing that slice in the `defer` instead of having closures?

Reply:
Yep exactly.